clap-v3-utils/src/input_parsers/signer.rs

```diff
@@ -236,16 +236,36 @@ pub fn try_keypair_of(
     matches: &ArgMatches,
     name: &str,
 ) -> Result<Option<Keypair>, Box<dyn error::Error>> {
-    matches.try_contains_id(name)?;
-    Ok(keypair_of(matches, name))
+    if let Some(value) = matches.try_get_one::<String>(name)? {
+        if value == ASK_KEYWORD {
+            let skip_validation = matches.try_contains_id(SKIP_SEED_PHRASE_VALIDATION_ARG.name)?;
+            keypair_from_seed_phrase(name, skip_validation, true, None, true).map(Some)
+        } else {
+            read_keypair_file(value).map(Some)
+        }
```

Comment:
It would be nice to de-deupe this with the logic in `try_keypairs_of` -- I know the duplication predates you.

Reply:
Yes good point. If `try_keypairs_of` returns a keypair vector of length greater than 1, it is actually unclear what the behavior of this function should be. Whether multiple keypairs are allowed should be caught by `Args`, so technically this choice will not matter if the function is used correctly... I pushed a change that simply pops the keypair out of the returned vector. Please take a look.

Reply:
Missing a push? I don't see any changes.

Reply:
Oh, sorry I don't quite know where those commit went... But I pushed the changes!

Reply:
Ohhh, I was confused about your comment until reading the code just now. By deduping, I actually meant put the following code in a helper fn that gets called from both methods:

```
        if value == ASK_KEYWORD {
            let skip_validation = matches.try_contains_id(SKIP_SEED_PHRASE_VALIDATION_ARG.name)?;
            keypair_from_seed_phrase(name, skip_validation, true, None, true).map(Some)
        } else {
            read_keypair_file(value).map(Some)
        }
```

Probably what you have written is fine, but I don't know if there might be costs to calling `try_get_many` when we only expect one?

Reply:
Oh oops! Sorry for the misinterpretation. I added a helper function `extract_keypair` to include the keypair extraction logic.

I think technically, the helper function should be named `keypair_from_path`, but we already have a public function with that name. The `try_keypair_of` and `try_keypairs_of` should probably use the existing `keypair_from_path` (unless we strictly want to restrict the source to just `ASK` or file), but that will be changing the functionality so we probably don't want to change it now.